client/modules/User/components/SignupForm.jsx

```diff
@@ -8,8 +8,13 @@ import Button from '../../../common/Button';
 import apiClient from '../../../utils/apiClient';
 
 function asyncValidate(fieldToValidate, value) {
+  if (fieldToValidate === 'email') {
+    return `Please enter an ${fieldToValidate}.`;
+  }
+
```

Comment:
Doesn't this need to be inside the `if (!value || value.trim().length === 0)` condition?  Otherwise it will return the message  even after the email has been entered.